src/test/java/com/sap/oss/phosphor/fosstars/data/github/CodeOfConductGuidelineInfoTest.java

```diff
@@ -0,0 +1,111 @@
+    GitHubProject project = new GitHubProject("test", "project");
+    LocalRepository localRepository = mock(LocalRepository.class);
+    when(localRepository.readTextFrom("CODE_OF_CONDUCT.md"))
+        .thenReturn(Optional.of(String.join("\n",
+            "Here is the code of conduct for our project.", "This is the text.", "Extra text")));
+    TestGitHubDataFetcher.addForTesting(project, localRepository);
+
+    CodeOfConductGuidelineInfo provider = new CodeOfConductGuidelineInfo(fetcher);
+
+    provider.requiredContentPatterns("Extra text");
+    ValueSet values = provider.fetchValuesFor(project);
+    checkValue(values, HAS_CODE_OF_CONDUCT, true);
+    checkValue(values, HAS_REQUIRED_TEXT_IN_CODE_OF_CONDUCT_GUIDELINE, true);
+
+    when(localRepository.readTextFrom("HOW_TO_CODE_OF_CONDUCT.md"))
+        .thenReturn(Optional.of(String.join("\n",
+            "Here is the code of conduct for our project.", "This is the text.")));
+
+    provider.knownCodeofConductGuidelineFiles("HOW_TO_CODE_OF_CONDUCT.md");
+    provider.requiredContentPatterns("Extra text.");
+    values = provider.fetchValuesFor(project);
+    checkValue(values, HAS_CODE_OF_CONDUCT, true);
+    checkValue(values, HAS_REQUIRED_TEXT_IN_CODE_OF_CONDUCT_GUIDELINE, false);
+  }
+
+  @Test
+  public void testLoadingConfig() throws IOException {
+    GitHubProject project = new GitHubProject("test", "project");
+    LocalRepository localRepository = mock(LocalRepository.class);
+    when(localRepository.readTextFrom("CODE_OF_CONDUCT.md"))
+         .thenReturn(Optional.of(String.join("\n",
+            "Here is how to contribute to the project.",
+            "Contributor Covenant",
+            "This is the Contributor Covenant")));
+    TestGitHubDataFetcher.addForTesting(project, localRepository);
+
+    CodeOfConductGuidelineInfo provider = new CodeOfConductGuidelineInfo(fetcher);
+    provider.configure(IOUtils.toInputStream(
+                    "---\n"
+                  + "requiredContentPatterns:\n"
+                  + "  - \"Contributor Covenant\""));
+
+    ValueSet values = provider.fetchValuesFor(project);
+    checkValue(values, HAS_CODE_OF_CONDUCT, true);
+    checkValue(values, HAS_REQUIRED_TEXT_IN_CODE_OF_CONDUCT_GUIDELINE, true);
+  }
+
+
+  @Test
+  public void testLoadingDefaultConfig() throws IOException {
+    Path config = Paths.get(String.format("%s.config.yml",
+        CodeOfConductGuidelineInfo.class.getSimpleName()));
+    String content =
+        "---\n"
+        + "requiredContentPatterns:\n"
+        + "  - \"Contributor Covenant\"";
+    Files.write(config, content.getBytes());
+    try {
+      CodeOfConductGuidelineInfo provider = new CodeOfConductGuidelineInfo(fetcher);
+      assertEquals(1, provider.requiredContentPatterns().size());
+      assertEquals(
+          "Contributor Covenant",
+          provider.requiredContentPatterns().get(0).pattern());
+    } finally {
+      FileUtils.forceDeleteOnExit(config.toFile());
+    }
+  }
+
```

Comment:
Can you add a test here?
- Load CodeOfCOnduct.config.yml
- Load Contributing.config.yml

- Assert when CodeOfConfuct provider has the appropriate Content patterns
- Assert when Contributing provider has the appropriate Content patterns

Also both the patterns will have the same attribute `requiredContentPatterns`

Reply:
My mistake. I over-engineered.
I had a look, looks like each Data provider loads a new set of configs required as per the name. Hence this is good for now.

Thank you for the changes.